tracker_package/tracker/src/tracker/tracker.py

```diff
@@ -0,0 +1,122 @@
+import rospy
+import select
+import cv2
+import numpy as np
+import socket
+import sys
+import pickle
+import struct
+import torch
+
+from ltr.data.bounding_box_utils import masks_to_bboxes
+from pytracking.evaluation import Tracker
+from sensor_msgs.msg import Image
+from std_msgs.msg import Float32MultiArray
+from cv_bridge import CvBridge, CvBridgeError
+from tracker.optical_flow import VisualTrackerKLT   
+
+
+class ObjectTracker:
+
+    def __init__(self):
+        self.clientsocket = socket.socket(socket.AF_INET,socket.SOCK_STREAM)
+        try:
+            self.clientsocket.connect((rospy.get_param('tracker/ip','172.17.0.1'), int(rospy.get_param("tracker/port",5556))))
+        except Exception as e:
+            self.clientsocket = None
+        self.run_optical_flow = bool(int(rospy.get_param('tracker/run_optical_flow', 0)))
+        self.tracker_run_iter =  int(rospy.get_param('tracker/track_run_iter', 10))
+        self.init_timeout = int(rospy.get_param('tracker/init_timeout', 20))
+        self.tracker_counter = 0
+        rospy.init_node("tracker")
+        rospy.Subscriber("image", Image, self.image_callback, queue_size=1)
+        self.bb_pub = rospy.Publisher("bounding_box", Float32MultiArray, queue_size=10)
+        self.tracker = Tracker("dimp", "dimp18")
+        self.init = False
+        self.of = VisualTrackerKLT()
+        self.bridge = CvBridge()
+
+    def image_callback(self, image_msg):
+        if self.clientsocket is None:
+            self.reconnect_to_socket()
+        if self.tracker_counter > 100:
+            self.tracker_counter = 0
+        self.tracker_counter += 1
+        modify_image = self.bridge.imgmsg_to_cv2(image_msg, "bgr8")
+        orig = modify_image.copy()
+        if not self.init:
+            if self.clientsocket is None:
+                return 
+            bounding_box = self.init_bounding_box(orig)
+            if self.run_optical_flow:
+                self.of.init(orig, bounding_box) 
+        
+        optical_flow_output = None
+        if self.run_optical_flow:
+            optical_flow_output = self.of(orig)
+            if optical_flow_output is not None:
+                min_x, min_y, max_x, max_y = optical_flow_output
```

Comment:
This can be done more nicely in several ways. Use one according to your preference.

Reply:
what you suggest?